src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -0,0 +1,363 @@
+                        OrchestrationRuntimeStatus.Completed,
+                        instance.InstanceId);
+#endif
+                    await this.LifeCycleNotificationHelper.OrchestratorCompletedAsync(
+                        this.Options.HubName,
+                        functionName.Name,
+                        instance.InstanceId,
+                        context.ContinuedAsNew,
+                        isReplay: false);
+                }
+                else
+                {
+                    this.TraceHelper.FunctionAwaited(
+                        this.Options.HubName,
+                        functionName.Name,
+                        FunctionType.Orchestrator,
+                        instance.InstanceId,
+                        isReplay: false);
+                }
+            }
+            else
+            {
+                string exceptionDetails = functionResult.Exception.ToString();
+
+                this.TraceHelper.FunctionFailed(
+                    this.Options.HubName,
+                    functionName.Name,
+                    instance.InstanceId,
+                    exceptionDetails,
+                    FunctionType.Orchestrator,
+                    isReplay: false);
+
+                await this.LifeCycleNotificationHelper.OrchestratorFailedAsync(
+                    this.Options.HubName,
+                    functionName.Name,
+                    instance.InstanceId,
+                    exceptionDetails,
+                    isReplay: false);
+
+                orchestratorResult = OrchestratorExecutionResult.ForFailure(
+                    message: $"Function '{functionName}' failed with an unhandled exception.",
+                    functionResult.Exception);
+            }
+
+            // Send the result of the orchestrator function to the DTFx dispatch pipeline.
+            // This allows us to bypass the default, in-process execution and process the given results immediately.
+            dispatchContext.SetProperty(orchestratorResult);
+        }
+
+        /// <summary>
+        /// Durable Task Framework activity middleware that invokes an out-of-process orchestrator function.
+        /// </summary>
+        /// <param name="dispatchContext">This middleware context provided by the framework that contains information about the activity.</param>
+        /// <param name="next">The next middleware handler in the pipeline.</param>
+        /// <exception cref="SessionAbortedException">Thrown if there is a recoverable error in the Functions runtime that's expected to be handled gracefully.</exception>
+        public async Task CallActivityAsync(DispatchMiddlewareContext dispatchContext, Func<Task> next)
+        {
+            TaskScheduledEvent? scheduledEvent = dispatchContext.GetProperty<TaskScheduledEvent>();
+            if (scheduledEvent == null)
+            {
+                // Not an activity
+                await next();
+                return;
+            }
```

Comment:
Same as above, how would we find ourselves in this method if not by invoking an activity?